Wikipedia/Code/SavedArticlesFetcher.m

```diff
@@ -140,9 +140,17 @@ - (void)fetchTitle:(MWKTitle*)title {
         self.fetchOperationsByArticleTitle[title] =
             [self.articleFetcher fetchArticleForPageTitle:title progress:NULL].thenOn(self.accessQueue, ^(MWKArticle* article){
             @strongify(self);
-            return [self downloadImageDataForArticle:article].thenOn(self.accessQueue, ^{
-                [self didFetchArticle:article title:title error:nil];
```

Comment:
@montehurd promises handled that… but basically we were ignoring errors so never implemented the catch block. Basically because this was a background job so there was nothing to show the user.